TestResultSummaryService/plugins/DataManagerPerf.js

```diff
@@ -0,0 +1,37 @@
+const child = require('../perf/DataManagerAggregate');
+
+module.exports.onBuildDone = async (task, { testResultsDB, logger }) => {
+    logger.debug("onBuildDone", task.buildName);
+    if ( !task.aggregateInfo ) {
+        let benchmarkName, benchmarkVariant, benchmarkProduct;
+        const metricsCollection = {};
+        if ( task.hasChildren ) {
+            // parent node.
+            const parentId = task._id;
+            const childBuildList = await testResultsDB.getData({parentId}).toArray();
+            //loop into the child build list
+            for ( let childBuild of childBuildList ) {
+                //loop into each child's tests info.
+                const {name, variant, product, benchmarkMetricsCollection} = await new child().aggDataCollect(childBuild)
+                benchmarkName = name;
+                benchmarkVariant = variant;
+                benchmarkProduct = product;
```

Comment:
- This three variables needed to be used in the following update async function outside of the scope of this loop. I have updated some lines to avoid duplicates. Please review.
- yes, actually for parent build, the only data source for this three variables are from its children builds, which means at least we need to go into the childrenBuildList loop to get it. I believe the design would make this three variables' data the same even from different children.So it doesn't matter which one we would like to keep. But this time I also add a continuity check this time for just in case.